src/contact-form.js

```diff
@@ -55,7 +99,23 @@ export default class ContactForm {
     const { modalRoot, elements } = this.getElements();
     const { onModalOpen, onFormSuccess, onFormFail } = this.options;
 
-    modalRoot.on('shown.bs.modal', () => elements[0].focus());
+    modalRoot.on('shown.bs.modal', () => {
+      this.modalIsOpen = true;
+      elements[0].focus();
+      if (this.xhrIsDone && !this.wasRendered) {
+        this.triggerElement.dispatchEvent(new CustomEvent('tryUpdateUI', {
```

Comment:
Minor detail: event name seems to be an action. You can name it `readyToUpdate` or something similar.
Additionally, you can add a boolean method with the following condition:

`this.xhrIsDone && !this.wasRendered` 

Seems  to be used in more than one place